http/src/main/resources/reference.conf

```diff
@@ -442,6 +442,16 @@ contentSecurityPolicy {
   ]
 }
 
+refererConfig {
+  # list of valid ports and hosts is environment specific so it's overriden in firecloud develop
+  validPorts = [
```

Comment:
Reading the code, it looks like `validPorts` is only used if the host == `"localhost"`. Wondering if that might be confusing to remember -- could we just represent the allow-list as host:port strings?
```
refererConfig {
  allowedHosts = [
    "localhost:3000",
    "app.terra.bio"
  ]
}
```

Then we could probably check the referer against it with `referer.getUri.authority.toString`:
```
scala> val uri1 = Uri("https://app.terra.bio/")
val uri1: akka.http.scaladsl.model.Uri = https://app.terra.bio/

scala> uri1.authority.toString
val res13: String = app.terra.bio

scala> val uri2 = Uri("http://localhost:3000")
val uri2: akka.http.scaladsl.model.Uri = http://localhost:3000

scala> uri2.authority.toString
val res14: String = localhost:3000
```

Reply:
Sounds great! Any approach where host:port can be determined in full (without possibility to spoof the check with something like `app.terra.bio.evil.org`) is good.